core/src/main/java/hudson/PluginFirstClassLoader.java

```diff
@@ -45,7 +48,21 @@
     extends AntClassLoader
     implements Closeable
 {
-    
+
+    private final Vector pathComponents;
+
+    public PluginFirstClassLoader() {
+        super();
+
+        try {
+            Field $pathComponents = AntClassLoader.class.getDeclaredField("pathComponents");
+            $pathComponents.setAccessible(true);
```

Comment:
Seems like you need to introduce a common ancestor with `AntClassLoader2` to avoid code duplication.